projects/pub_views.py

```diff
@@ -69,16 +68,25 @@ def user_publications(request):
         try:
             del_pub_id = request.POST["pub_ref"]
             logger.debug("deleting publication with id {}".format(del_pub_id))
-            Publication.objects.get(pk=del_pub_id).delete_pub()
+            pub = Publication.objects.get(pk=del_pub_id)
+            if pub.added_by_username != request.user.username:
+                messages.error(
+                    request,
+                    "You do not have permission to delete that publication!",
+                )
+            else:
+                pub.delete_pub()
         except Exception:
             logger.exception("Failed removing publication")
             messages.error(
                 request,
                 "An unexpected error occurred while attempting "
                 "to remove this publication. Please try again",
             )
+    mapper = ProjectAllocationMapper(request)
```

Comment:
This section now queries for publications for all of the user projects, not just publications the user has submitted.